File: backend/pkg/currency/currency.go
package currency

import "fmt"

func FormatUSD(amount float64) string {
	return fmt.Sprintf("$%.2f", amount)
}

func FormatKHR(amount float64) string {
	return fmt.Sprintf("\u17DB%.0f", amount)
}

func ConvertToKHR(usdAmount, exchangeRate float64) float64 {
	return usdAmount * exchangeRate
}

File: frontend/src/utils/currency.ts
export function formatUSD(amount: number): string {
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
  }).format(amount)
}

export function formatKHR(amount: number): string {
  return new Intl.NumberFormat('km-KH', {
    style: 'currency',
    currency: 'KHR',
    maximumFractionDigits: 0,
  }).format(amount)
}

export function convertToKHR(usdAmount: number, exchangeRate: number): number {
  return usdAmount * exchangeRate
}
